### PR TITLE
Add bgp neighbour metrics

### LINKED ISSUE #100: BGP metrics
I was playing around with the exporter and really like this project.

However when creating a dashboard I realised that I was missing some BGP metrics, more specifically:
* BPG peer state
* Routes received and advertised (and if possible if the routes is used AKA. best path)

There might even be more that I'm not thinking about?

If you like the idea I can give it a shot implementing and if it works out creating a PR.

### PR DESCRIPTION
**Pull request by bl0m1**
This PR adds the metric `fortigate_bgp_neighbors` that will show all bgp (ipv4 and ipv6) neighbors :)

This metrics is only available in version 7.0.0 and later

Fixes #100

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -171,6 +172,7 @@ config system accprofile
             # If you do not wish to grant this permission, the relevant
             # labels/metrics will be absent.
             set cfg read
+            # If you wish to collect ipv6 bgp neighbours please add `set route-cfg read`
```
Comment (bluecmd) [inline README.md:175]:
```suggestion
            # If you wish to collect ipv6 bgp neighbours, add this:
            set route-cfg read
```

File: pkg/probe/bgp_neighbors_ipv4.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+	var (
+		bpgNeighbor = prometheus.NewDesc(
+			"fortigate_bgp_neighbors_ipv4",
+			"Confiured bgp neighbors over ipv4",
+			[]string{"vdom", "remote_as", "state", "admin_status", "local_ip", "neighbor_ip", "type"}, nil,
+		)
+	)
+
+	type BgpNeighbor struct {
```
Comment (bluecmd) [inline pkg/probe/bgp_neighbors_ipv4.go:23]:
```suggestion
	type BGPNeighbor struct {
```

File: pkg/probe/bgp_neighbors_ipv4.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
+package probe
```
Comment (bluecmd) [inline pkg/probe/bgp_neighbors_ipv4.go:1]: Maybe keep ipv4 and ipv6 in the same file? Seems a bit much to split them up given it is only one function in the file.

File: pkg/probe/bgp_neighbors_ipv4.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	"github.com/bluecmd/fortigate_exporter/pkg/http"
+	"github.com/bluecmd/fortigate_exporter/internal/version"
+)
+
+func probeBgpNeighborsIPv4(c http.FortiHTTP) ([]prometheus.Metric, bool) {
+
+	var (
+		bpgNeighbor = prometheus.NewDesc(
+			"fortigate_bgp_neighbors_ipv4",
+			"Confiured bgp neighbors over ipv4",
```
Comment (bluecmd) [inline pkg/probe/bgp_neighbors_ipv4.go:18]:
```suggestion
			"State of configured IPv4 BGP peers",
```

File: pkg/probe/bgp_neighbors_ipv4.go [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+			[]string{"vdom", "remote_as", "state", "admin_status", "local_ip", "neighbor_ip", "type"}, nil,
```
Comment (bluecmd) [inline pkg/probe/bgp_neighbors_ipv4.go:19]: Type is redundant, it is only `ipv4` or `ipv6` which is implied by the `_ipv4` metric name.

File: pkg/probe/bgp_neighbors_ipv4.go (hunk 3 above)
Comment (bluecmd) [reply pkg/probe/bgp_neighbors_ipv4.go:1]: In that case you can also share the BGPNeighbor structure (unless they differ?) between IPv4 and IPv6.

File: pkg/probe/bgp_neighbors_ipv4.go [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+	"log"
+	"strconv"
+    
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/bluecmd/fortigate_exporter/pkg/http"
+	"github.com/bluecmd/fortigate_exporter/internal/version"
+)
+
+func probeBgpNeighborsIPv4(c http.FortiHTTP) ([]prometheus.Metric, bool) {
```
Comment (bluecmd) [inline pkg/probe/bgp_neighbors_ipv4.go:13]:
```suggestion
func probeBGPNeighborsIPv4(c http.FortiHTTP) ([]prometheus.Metric, bool) {
```

**Review by bluecmd — CHANGES_REQUESTED** [review]

File: pkg/probe/bgp_neighbors_ipv4.go (hunk 3 above)
Comment (bl0m1) [reply pkg/probe/bgp_neighbors_ipv4.go:1]: Agree, I'll combined them into one file

File: README.md [hunk 7]
```diff
@@ -57,6 +57,7 @@ Per-VDOM:
  * `fortigate_ipsec_tunnel_receive_bytes_total`
  * `fortigate_ipsec_tunnel_transmit_bytes_total`
  * `fortigate_ipsec_tunnel_up`
+ * `fortigate_bgp_neighbors`
```
Comment (secustor) [inline README.md:60]: Please rename the metrics to `fortigate_bgp_neighbor_ipv4_info` and  `fortigate_bgp_neighbor_ipv6_info`   
As `_info` signals that this is an info metric and contains the information in the labels and use singular as the information is about a single bgp neighbor.

File: pkg/probe/bgp_neighbors_ipv4.go [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+		NeighborIP        string       `json:"neighbor_ip"`
+		LocalIP           string       `json:"local_ip"`
+		RemoteAS          int          `json:"remote_as"`
+		AdminStatus       bool         `json:"admin_status"`
+		State             string       `json:"state"`
+		Type              string       `json:"type"`
+	}
+
+	type BpgNeighborResponse struct {
+		HTTPMethod string                 `json:"http_method"`
```
Comment (secustor) [inline pkg/probe/bgp_neighbors_ipv4.go:33]: Please remove unused struct fields.

File: pkg/probe/bgp_neighbors_ipv6.go [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+		NeighborIP        string       `json:"neighbor_ip"`
+		LocalIP           string       `json:"local_ip"`
+		RemoteAS          int          `json:"remote_as"`
+		AdminStatus       bool         `json:"admin_status"`
+		State             string       `json:"state"`
+		Type              string       `json:"type"`
+	}
+
+	type BpgNeighborResponse struct {
+		HTTPMethod string                 `json:"http_method"`
```
Comment (secustor) [inline pkg/probe/bgp_neighbors_ipv6.go:33]: Remove here also the unused fields.

**Review by secustor — CHANGES_REQUESTED** [review]
Some additional remarks, but nothing big.

**Review by bluecmd — APPROVED** [review]
LGTM @secustor any feedback?

File: README.md (hunk 7 above)
Comment (bl0m1) [reply README.md]: Agreed, fixed :)

File: pkg/probe/bgp_neighbors.go [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+	LocalIP           string       `json:"local_ip"`
+	RemoteAS          int          `json:"remote_as"`
+	AdminStatus       bool         `json:"admin_status"`
+	State             string       `json:"state"`
+}
+
+type BGPNeighborResponse struct {
+	Results    []BGPNeighbor          `json:"results"`
+	VDOM       string                 `json:"vdom"`
+	Path       string                 `json:"path"`
```
Comment (secustor) [inline pkg/probe/bgp_neighbors.go:24]: Please remove the unused fields `path`, `name`, `status` and `build`

File: pkg/probe/bgp_neighbors_test.go [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+func TestBGPNeighborsIPv4(t *testing.T) {
+	c := newFakeClient()
+	c.prepare("api/v2/monitor/router/bgp/neighbors", "testdata/bpg-neighbors-v4.jsonnet")
+	r := prometheus.NewPedanticRegistry()
+	if !testProbe(probeBGPNeighborsIPv4, c, r) {
+		t.Errorf("probeBGPNeighborsIPv4() returned non-success")
+	}
+
+	em := `
+	# HELP fortigate_bgp_neighbor_ipv4_info Confiured bgp neighbor over ipv4
```
Comment (secustor) [inline pkg/probe/bgp_neighbors_test.go:21]: Why are the test lines differently indented?

**Review by secustor — CHANGES_REQUESTED** [review]

File: pkg/probe/bgp_neighbors_test.go (hunk 11 above)
Comment (bl0m1) [reply pkg/probe/bgp_neighbors_test.go]: Fixed, this was missed by my editor

File: pkg/probe/bgp_neighbors.go (hunk 10 above)
Comment (bl0m1) [reply pkg/probe/bgp_neighbors.go]: Fixed :)

**Review by secustor — APPROVED** [review]
LGTM  Thanks for your contribution!

**Pull request merged**